packages/front-end/services/DefinitionsContext.tsx

```diff
@@ -308,3 +310,13 @@ export const DefinitionsProvider: FC<{ children: ReactNode }> = ({
     </DefinitionsContext.Provider>
   );
 };
+
+export function DefinitionsGuard({ children }: { children: ReactElement }) {
+  const { ready, error } = useDefinitions();
+
+  if (!error && !ready) {
+    return <LoadingOverlay />;
+  }
```

Comment:
Is it worth also abstracting out the error case here so it doesn't need to be handled by each child component? (Doesn't have to be in this PR)